specs/howto.md

```diff
@@ -0,0 +1,349 @@
+
+This document aims to provide few simple guidelines for the adoption of FIWARE
+Harmonised Data Models. This guide is not exhaustive and does not aim to
+cover the specifics of each model, rather it provides general usage tips valid
+for most of the existing models and for expected models in the future.
+
+Data Models have been defined in relation to the FIWARE reference data context
+model ([NGSI v2](http://fiware.github.io/specifications/ngsiv2/stable/)). While
+this does not imply that they cannot be used in a different context, however it
+does indicate that some of [design principles](guidelines.md) have been driven
+by that.
+This also implicates that Data Models will follow the evolution of FIWARE
+platform, NGSI-LD mappings will be added to all existing Data Models
+in the medium term.
+
+## General principles
+
+The general principle guiding your adoption should be: keep it simple.
+
+* **Use only the parts of the Data Model your application really need.** This will
+  help reducing the amount of information exchanged to the one you need and not
+  more than that. This may apply also just to a part of your application. I.e.
+  while you use in your Context Broker the whole Data Model, to other services
+  you may dispatch only the needed part.
+* **Use metadata only in case of need, and follow the assumptions provided in the
+  Data Model.** For example, in case your application is leveraging the default
+  unit defined in a Data Model (e.g. meters), avoid attaching metadata to
+  annotate the unit. In case your application is using a different unit,
+  do annotate it, it will simplify data exchange with other systems in case of
+  need.  
+* **Stick to the existing Data Models.** Data Models are meant to be an effort
+  to facilitate interoperability within the community. If you change the meaning
+  or the data type attached to an attribute, this will broke interoperability.
+  Of course it may be the case that the model is not covering your scenario,
+  or it contains errors. Open a ticket in the repository, and let the community
+  discuss on the issue!
+* **While not generally advised, you can add additional attributes in your
+  specific application for the purpose of your application.** Be aware that of
+  course this means that such attributes may not be correctly used by other
+  applications leveraging the same Data Model. We recommend, if the attribute
+  you want to add is of general adoption and not specific to your case, to
+  propose it to become officially part of the Data Model. You can open an issue
+  or a pull request for that.
+* **Data Models, in most of the cases, are NGSI v2 codification of existing data
+  models.** You can find the references in each Data Model specification, but
+  for the sake of maintainability, on purpose we don't report the full
+  specification. This means that to gather further details, you need to access
+  the original source. For example, for the location, we use GeoJSON. You
```

Comment:
Nice use of the example, until the example it was not clear to me what this point was talking about :)